Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace rogue.src.sim
+{
+    class EntityComponents
+    {
+        private Entity Entity;
+
+        private Dictionary<Type, Component> Components = new Dictionary<Type, Component>();
+
+        public EntityComponents(Entity entity)
+        {
+            Entity = entity;
+        }
+
+        public Component Get<TComponent>()
+            where TComponent : Component
+        {
+            return Components[typeof(TComponent)];
+        }
+
+        public Component Get(Type componentType)
+        {
+            return Components[componentType];
+        }
+
+        public Component Attach<TComponent>()
+            where TComponent : Component, new()
+        {
+            if (Components.ContainsKey(typeof(TComponent)))
+            {
+                return Components[typeof(TComponent)];
+            }
+
+            var newComponent = new TComponent();
+            Components[typeof(TComponent)] = newComponent;
+            newComponent.HandleAttachToEntity(Entity);
+            return newComponent;
+        }
+
+        public Component Attach(Type type)
+        {
+            if (Components.ContainsKey(type))
+            {
+                return Components[type];
+            }
+
+            Component newComponent = Activator.CreateInstance(type) as Component;
+            if (newComponent == null) { return null; }
```

Comment:
Yikes; does this happen? From what I can tell in the docs, `Activator.CreateInstance` shouldn't return null; it should throw an exception if we fuck up.